https support

rubdos opened the pull request:
Will eventually resolve #369.

rubdos committed:
add https option

Jalle19 commented on src/tvheadend/HTSPConnection.cpp:
```diff
@@ -144,7 +144,13 @@ std::string HTSPConnection::GetWebURL ( const char *fmt, ... ) const
     auth += "@";
 
-  std::string url = StringUtils::Format("http://%s%s:%d", auth.c_str(), settings.GetHostname().c_str(), settings.GetPortHTTP());
+  const char *proto = nullptr;
```
Could be shortened to `const std::string proto = settings.GetUseHTTPS() ? 'https' : 'http';`

rubdos commented:
I haven 't tested this yet, will do that later... Just wanted this out somewhere already :)

Jalle19 commented on src/tvheadend/Settings.h:
```diff
@@ -106,6 +107,7 @@ namespace tvheadend {
     : m_strHostname(DEFAULT_HOST),
       m_iPortHTSP(DEFAULT_HTTP_PORT),
       m_iPortHTTP(DEFAULT_HTSP_PORT),
+      m_buseHTTPS(DEFAULT_USE_HTTPS),
```
Wrong member name here, should be `m_bUseHTTPS`

rubdos replied:
I should really compile this before pushing... :'-)

ksooo commented on src/tvheadend/Settings.h:
```diff
@@ -38,6 +38,7 @@ namespace tvheadend {
     static const std::string DEFAULT_HOST;
     static const int         DEFAULT_HTTP_PORT;
     static const int         DEFAULT_HTSP_PORT;
+    static const bool         DEFAULT_USE_HTTPS;
```
wrong indentation

rubdos replied:
Slip of the eye. It compiles now, but I've not been able to test yet. Gotta get something x86 with X11 for that.

rubdos commented:
Mind that I still didn't have a chance to test this, so please do so before merging :'-)

rubdos commented:
I tested it working, but the text for the button in the settings is still empty.

rubdos commented:
> I tested it working, but the text for the button in the settings is still empty.

Could someone give me a hint how I can update the necessary translation files for this?

Gee1111 commented:
this also works for the windows addon ?

rubdos commented:
> this also works for the windows addon ?

I wouldn't know how that'd be different. But hey, it's Windows, better test it :-)

Jalle19 commented:
> Could someone give me a hint how I can update the necessary translation files for this?

You only need to update en_GB, the rest are handled later on through Transifex. If you're using another language in Kodi you might wanna switch to the en_GB while developing so you can see your changes.

ksooo commented on pvr.hts/resources/settings.xml:
```diff
@@ -2,6 +2,7 @@
 <settings>
   <category label="30000">
     <setting id="host" type="text" label="30001" default="127.0.0.1" />
+    <setting id="https" type="bool" label="30018" default="false" />
```
Please use id 30011 and add a string for it here https://github.com/kodi-pvr/pvr.hts/blob/master/pvr.hts/resources/language/resource.language.en_gb/strings.po#L65

rubdos replied:
Done.

ksooo reviewed (changes requested):
please change settings id to first available general settings id (30011) and supply a string for it.

ksooo commented:
@rubdos have you seen that we requested changes?

rubdos commented:
Excuse me for the late reply. Have been quite busy.

W.r.t. base branch: are you sure we don't want to backport this into Leia? It's quite an annoying bug imo.

jobrien2001 commented:
@ksooo Any chance to review this? :D

rubdos commented:
Apparently I never fixed the translation ID. That's done now. I also rebased against Matrix.

trimethia commented:
@ksooo Any chance to review this?

AlwinEsch commented:
Related to https://github.com/xbmc/xbmc/pull/17307, not forget a rebase of request to allow further Jenkins use!

```sh
git remote add upstream https://github.com/kodi-pvr/pvr.hts
git fetch upstream
git rebase upstream/Matrix
```

rubdos commented:
It's only the third time I'm rebasing this now, I would really appreciate to know what still has to be done to get this merged.

ksooo commented:
Please update the changelog and increase addon minor version and we are ready to go.

ksooo commented:
> Please update the changelog and increase addon minor version and we are ready to go.

 Well not really. Kodi default language is en_gb, not en_us. So, please add the new settings string to pvr.hts/resources/language/resource.language.en_gb/strings.po and remove it from pvr.hts/resources/language/resource.language.en_us/strings.po as   translations are handled via Transifex, not Github.

ksooo reviewed (changes requested):
See my comments

rubdos commented:
> See my comments

I don't see any comments in the diff linked by "View changes".

I've changed the translation to `en_gb`, added the changelog entry, incremented the minor version. Would you like me to squash everything together?

ksooo commented:
Yes, please squash to a single commit.

ksooo commented on pvr.hts/addon.xml.in:
```diff
@@ -1,7 +1,7 @@
 <addon
   id="pvr.hts"
-  version="5.3.4"
+  version="5.3.5"
```
Please increase the minor version number, not the micro as we are adding a new feature with this PR. => 5.4.0

ksooo commented on src/tvheadend/HTSPConnection.cpp:
```diff
@@ -149,7 +149,8 @@ std::string HTSPConnection::GetWebURL(const char* fmt, ...) const
     auth += "@";
 
-  std::string url = StringUtils::Format("http://%s%s:%d", auth.c_str(),
+  const char *proto = settings.GetUseHTTPS() ? "https" : "http";
```
Please left align the asterisk. Coding guide lines. => 'const char* proto'

rubdos commented:
Comments addressed, squashed!

ksooo reviewed (approved):
Thanks for your contribution.
